contractcourt/chain_arbitrator.go

```diff
@@ -242,6 +242,10 @@ func newActiveChannelArbitrator(channel *channeldb.OpenChannel,
 				return nil, err
 			}
 
+			if err := channel.MarkBorked(); err != nil {
+				return nil, err
+			}
+
```

Comment:
we should move the instantiation of the `chanMachine` until after the link has been removed from the switch, as it's possible it can be mutated while the link is shutting down

Reply:
currently it's possible that it could read a commitment, which gets replaced while link is shutting down, then plays the old one

Reply:
Re-ordered, current order:
  * mark as borked
  * mark link inactive
  * fetch channel (peer not there anymore)
  * force close the channel

Reply:
nice, ordering lgtm